src/cargo/util/command_prelude.rs

```diff
@@ -238,57 +237,45 @@ pub trait AppExt: Sized {
     }
 
     fn arg_quiet(self) -> Self {
-        self._arg(opt("quiet", "Do not print cargo log messages").short("q"))
+        self._arg(opt("quiet", "Do not print cargo log messages").short('q'))
     }
 }
 
 impl AppExt for App {
-    fn _arg(self, arg: Arg<'static, 'static>) -> Self {
+    fn _arg(self, arg: Arg<'static>) -> Self {
         self.arg(arg)
     }
 }
 
-pub fn opt(name: &'static str, help: &'static str) -> Arg<'static, 'static> {
-    Arg::with_name(name).long(name).help(help)
+pub fn opt(name: &'static str, help: &'static str) -> Arg<'static> {
+    Arg::new(name).long(name).help(help)
 }
 
-pub fn optional_opt(name: &'static str, help: &'static str) -> Arg<'static, 'static> {
+pub fn optional_opt(name: &'static str, help: &'static str) -> Arg<'static> {
     opt(name, help).min_values(0)
 }
 
 pub fn optional_multi_opt(
     name: &'static str,
     value_name: &'static str,
     help: &'static str,
-) -> Arg<'static, 'static> {
+) -> Arg<'static> {
     opt(name, help)
         .value_name(value_name)
-        .multiple(true)
+        .multiple_occurrences(true)
+        .multiple_values(true)
         .min_values(0)
         .number_of_values(1)
 }
```

Comment:
Didn't dig too deeply into how this works, it might be possible to simplify it.

Reply:
`optional_multi_opt` is supposed to allow `cargo cmd --theopt value1 --otheropt` and `cargo cmd --theopt value1 --theopt value2 --otheropt` and `cargo cmd --theopt --otheropt`, but not `cargo cmd --theopt value1 value2 --otheropt`.

Reply:
So its like `--color[=WHEN]` with multiple.

What confused me is I never realized these args work like this,  Playing with some of them it now makes sense; at least in some of the cases, like `cargo check --test`, its so cargo can list what values are supported.

Reply:
Also, as far as I can tell, because cargo accepts optional values passed to options, and doesn't require `=` on those options, we've effectively ruled out the possibility of having positional parameters on various commands.